File: lick.py
class Lick:
    """
    Stores a lick with notes relative to a root chord.

    Each element in note specifies a beat. Sublists are used to divide the beat.
    """
    def __init__(self, chord, *notes):
        self.__chord = chord
        self.__notes = notes

    def __iter__(self):
        for beat in self.__notes:
            if isinstance(beat, list):
                for note in beat:
                    yield note
            else:
                yield (beat, 1)

licks = (
    Lick(1, (1, 0), (1, 3)),
    Lick(1, (3, 4), (6, 7))
)

File: c/lick.c
#include "beg.h"

bool licks_generator(struct lick *l) {
    static i = 0;
    if(i >= NUM_LICKS) {
        i = 0;
    }

    *l = licks[i++];
    return i < NUM_LICKS;
}

bool notes_generator(struct lick l, struct note *n) {
    static _i = 0;
    if(n->notes->dur == 0) {
        i = 0;
    }

    *n = licks + (lick * LICK_LENGTH) + 1 + i;
    i++;

    return n->note != 0;
}